### PR TITLE
Chore: Embed CSV files with the test data source

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Remove the dependency of the Grafana settings from the test data source. The only place in which this configuration was read was to get the `StaticRootPath` which was used to read CSV files that should be packed with the plugin. Now the plugin uses [Go embed](https://pkg.go.dev/embed) to include the CSV files with the binary so it's not needed to read external files.

This PR also touches some files that contained references to the previous path of the `testdata` CSV files.

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/71329

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tsdb/testdatasource/csv_data.go [hunk 1]
```diff
@@ -92,6 +92,9 @@ func (s *Service) handleCsvFileScenario(ctx context.Context, req *backend.QueryD
 	return resp, nil
 }
 
+//go:embed plugindata/*.csv
```
Comment (wbrowne) [inline pkg/tsdb/testdatasource/csv_data.go:95]: Nit: Would probably just call this `data` instead of `plugindata`

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -444,7 +444,6 @@ lerna.json @grafana/frontend-ops
 /public/robots.txt @grafana/frontend-ops
 /public/sass/ @grafana/grafana-frontend-platform
 /public/test/ @grafana/grafana-frontend-platform
-/public/testdata/ @grafana/grafana-frontend-platform
```
Comment (wbrowne) [inline .github/CODEOWNERS:447]: Is this deleted? 🤔

**Review by wbrowne — APPROVED** [review]
LGTM

**Review by xnyo — APPROVED** [review]
LGTM!

File: .github/CODEOWNERS (hunk 2 above)
Comment (xnyo) [reply .github/CODEOWNERS:447]: Yes the folder doesn't exist anymore:
https://github.com/grafana/grafana/tree/f6b9f37839a153403c209cde6a44bdef6eb73121/public

Comment (andresmgot) [reply .github/CODEOWNERS:447]: correct, this folder only contained the CSV files for the test data source.

**Review by ryantxu — APPROVED** [review]

**Pull request merged**